src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -72,25 +72,30 @@ end
 function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
     F = cholesky(X)
     function cholesky_pullback(Ȳ)
-        ∂X = @thunk(chol_blocked_rev(Matrix(Ȳ), Matrix(F.U), 25, true))
+        ∂X = if F.uplo === 'U'
+            @thunk(chol_blocked_rev(Matrix(Ȳ.U), Matrix(F.U), 25, true))
+        else
+            @thunk(chol_blocked_rev(Matrix(Ȳ.L), Matrix(F.L), 25, false))
```

Comment:
I do not know why we need to call `Matrix` here (these are `UpperTriangular` or `LowerTriangular` matrices)

Reply:
scratch that --  added a comment. Otherwise we eventually end up with a SubArray passed to `trsm!` and that is not a "StridedArray" and blas complains
```
ERROR: MethodError: no method matching strides(::UpperTriangular{Float64,Array{Float64,2}})
```